providers/universal-provider/src/providers/eip155.ts

```diff
@@ -248,13 +273,55 @@ class Eip155Provider implements IProvider {
       }
     }
 
+    const storedSendCalls = await this.getStoredSendCalls(args.request.params?.[0] as string);
+    if (storedSendCalls) {
+      try {
+        return await this.prepareCallStatusFromStoredSendCalls(storedSendCalls);
+      } catch (error) {
+        console.warn("Failed to fetch call status from stored send calls", error, storedSendCalls);
+      }
+    }
+
     if (this.namespace.methods.includes(args.request.method)) {
       return await this.client.request(args as EngineTypes.RequestParams);
     }
 
     throw new Error("Fetching call status not approved by the wallet.");
   }
 
+  private async prepareCallStatusFromStoredSendCalls(storedSendCalls: StoredSendCalls) {
+    const chainId = parseChainId(storedSendCalls.result.capabilities.caip345.caip2);
+    const hashes = storedSendCalls.result.capabilities.caip345.transactionHashes;
+    const allPromises = await Promise.allSettled(
+      hashes.map((hash) => this.getTransactionReceipt(chainId.reference, hash)),
+    );
+
+    const receipts = allPromises.filter((r) => r.status === "fulfilled").map((r) => r.value);
+
+    // log failed transactions
+    allPromises
+      .filter((r) => r.status === "rejected")
+      .forEach((r) => console.warn("Failed to fetch transaction receipt:", r.reason));
+
+    const allReceiptsSuccessful = receipts.every((r) => r.status === "0x1");
+
+    return {
```

Comment:
We're missing chainId in the response: https://eips.ethereum.org/EIPS/eip-5792#wallet_getcallsstatus-rpc-specification